src/com/xilinx/rapidwright/rwroute/GlobalSignalRouting.java

```diff
@@ -0,0 +1,855 @@
+	 * @param unavailableNodes A set of unavailable nodes.
+	 * @param visitedRoutingNodes
+	 * @return true, if the RoutingNode should not be considered as an available resource.
+	 */
+	private static boolean pruneNode(RoutingNode routingNode, Set<Node> unavailableNodes, Set<RoutingNode> visitedRoutingNodes){
+		Node n = routingNode.getNode();
+		IntentCode ic = n.getTile().getWireIntentCode(n.getWire());
+		switch(ic){
+			case NODE_GLOBAL_VDISTR:
+			case NODE_GLOBAL_HROUTE:
+			case NODE_GLOBAL_HDISTR:
+			case NODE_HLONG:
+			case NODE_VLONG:
+			case NODE_GLOBAL_VROUTE:
+			case NODE_GLOBAL_LEAF:
+			case NODE_GLOBAL_BUFG:
+				return true;
+			default:
+		}
+		if(unavailableNodes.contains(n)) return true;
+		if(visitedRoutingNodes.contains(routingNode)) return true;
+		return false;
+	}
+	
+	/**
+	 * Determines if the given RoutingNode can serve as our sink.
+	 * @param routingNode The RoutingNode in question.
+	 * @param type The net type to designate the static source type.
+	 * @return true if this sources is usable, false otherwise. 
+	 */
+	private static boolean isThisOurStaticSource(Design design, RoutingNode routingNode, NetType type, Set<RoutingNode> usedRoutingNodes){
+		if(usedRoutingNodes != null && usedRoutingNodes.contains(routingNode))
+			return true;
+		Node node = routingNode.getNode();
+		return isNodeUsableStaticSource(node, type, design);
+	}
+	
+	/**
+	 * This method handles queries during the static source routing process. 
+	 * It determines if the node in question can be used as a source for the current NetType.
+	 * @param routingNode The ResourceNode in question.
+	 * @param type The NetType to indicate what kind of static source we need (GND/VCC).
+	 * @return True if the pin is a hard source or an unused LUT output that can be repurposed as a source.
+	 */
+	private static boolean isNodeUsableStaticSource(Node node, NetType type, Design design){
+		// We should look for 3 different potential sources
+		// before we stop:
+		// (1) GND_WIRE 
+		// (2) VCC_WIRE 
+		// (3) Unused LUT Outputs (A_0, B_0,...,H_0)
+		String pinName = type == NetType.VCC ? Net.VCC_WIRE_NAME : Net.GND_WIRE_NAME;		
+		if(node.getWireName().startsWith(pinName)){
+			return true;
+		}else if(lutOutputPinNames.contains(node.getWireName())){
+			Site slice = node.getTile().getSites()[0];
+			SiteInst i = design.getSiteInstFromSite(slice);			
+			if(i == null) return true; // Site is not used
+			char uniqueId = node.getWireName().charAt(node.getWireName().length()-3);
+			Net currNet = i.getNetFromSiteWire(uniqueId + "_O");
+			if(currNet == null) return true;
+			if(currNet.getType() == type) return true;
+			return false;
+		}
+		return false;
+	}
+	
+	/**
+	 * A lightweight useful class for different simple routing-related scenarios, 
+	 * each {@link RoutingNode} Object is associated to a {@link Node} Object.
+	 */
+	static class RoutingNode{
+		private Node node;
+		private RoutingNode prev;
+		private boolean isTareget;
```

Comment:
```suggestion
		private boolean isTarget;
```
(and also its uses and methods below)